### ISSUE TITLE
Enable multiple hash functions

### ISSUE THREAD

**Issue by ctb** [dib-lab/khmer#27]
Right now, we have only one DNA hash function, and it's a perfect hash that goes from (up to a) 32-base DNA string into a 64-bit number, to which we then apply a modulus. It also maps the forward and reverse complement hashes consistently to the lower of the two, which means that AAAAA and TTTTT always hash to the same number.  Changing this rev comp behavior requires recompilation, and in general there is no way to change the hash function being used without recompiling khmer.

We would like to be able to use Jordan's cyclic hash implementation and also enable single-stranded DNA/RNA hashing without recompilation.

One suggestion from CTB is to allow Hashtable objects to have their own hashing function.  It would be interesting to explore the impact of this on the actual lib and script code.  It would also be interesting to see how big the performance hit is of doing this.

**Comment by ctb** [comment]
Also relevant: #60, k > 32, being implemented in #624.

**Comment by ctb** [comment]
The cityhash library (https://code.google.com/p/cityhash/) might be worth looking at.

**Comment by luizirber** [comment]
"CityHash provides hash functions for strings. It has been superseded by [FarmHash][2]. Further updates to CityHash are fairly unlikely."

There is also [xxHash][1].

[1]: https://github.com/Cyan4973/xxHash
[2]: https://code.google.com/p/farmhash/

**Comment by betatim** [comment]
Another hash function: https://github.com/lemire/rollinghashcpp I spied some "ACGT" strings in an example so the author might have had genomics in mind. Unsure how battle tested it is.

**Comment by ctb** [comment]
Multiple hash functions are enabled by #1511.

**Comment by ctb** [comment]
Lots of good hash functions mentioned in here, but I'm closing because after #1511 merge we can start actually playing with them (e.g. #1622).
